### PR TITLE
Update to sygus standard output for check-synth responses

### PR DESCRIPTION
**Pull request by ajreynol**
This PR does two things:
(1) It eliminates the ad-hoc implementation of printSynthSolutions and removes it from the API.  Now, printing response to a check-synth command is done in a more standard way, using the API + symbol manager.  This is analogous to recent refactoring to get-model.
(2) It updates cvc5's output in response to check-synth to be compliant with the upcoming sygus 2.1 standard. The standard has changed slightly: responses to `check-synth` are now closed in parentheses, mirroring the smt2 response to `get-model`.

It also removes the unused command GetSynthSolutionCommand.

### CODE REVIEW

**Comment by ajreynol** [comment]
Decided to eliminate printSynthSolutions, @abdoo8080 will update the examples in a separate PR first.

File: src/expr/symbol_manager.cpp [hunk 1]
```diff
@@ -65,10 +66,14 @@ class SymbolManager::Implementation
   std::vector<api::Sort> getModelDeclareSorts() const;
   /** get model declare terms */
   std::vector<api::Term> getModelDeclareTerms() const;
+  /** get functions to synthesize */
+  std::vector<api::Term> getFunctionsToSynthesize() const;
   /** Add declared sort to the list of model declarations. */
   void addModelDeclarationSort(api::Sort s);
   /** Add declared term to the list of model declarations. */
   void addModelDeclarationTerm(api::Term t);
+  /** Add function to the list of functions-to-synthesize. */
```
Comment (HanielB) [inline src/expr/symbol_manager.cpp:75]:
```suggestion
  /** Add function to the list of functions to synthesize. */
```

File: src/expr/symbol_manager.cpp [hunk 2]
```diff
@@ -192,6 +199,13 @@ std::vector<api::Term> SymbolManager::Implementation::getModelDeclareTerms()
   return declareTerms;
 }
 
+std::vector<api::Term> SymbolManager::Implementation::getFunctionsToSynthesize()
+    const
+{
+  std::vector<api::Term> sfuns(d_funToSynth.begin(), d_funToSynth.end());
+  return sfuns;
```
Comment (HanielB) [inline src/expr/symbol_manager.cpp:206]: Can you inline this?

File: src/expr/symbol_manager.h [hunk 3]
```diff
@@ -120,6 +125,11 @@ class CVC5_EXPORT SymbolManager
    * Add declared term to the list of model declarations.
    */
   void addModelDeclarationTerm(api::Term t);
+  /**
+   * Add a function-to-synthesize. This ensures the solution for f is printed
```
Comment (HanielB) [inline src/expr/symbol_manager.h:129]:
```suggestion
   * Add a function to synthesize. This ensures the solution for f is printed
```

File: src/smt/command.cpp [hunk 4]
```diff
@@ -699,6 +699,7 @@ const api::Grammar* SynthFunCommand::getGrammar() const { return d_grammar; }
 
 void SynthFunCommand::invoke(api::Solver* solver, SymbolManager* sm)
 {
+  sm->addFunctionToSynthesize(d_fun);
```
Comment (HanielB) [inline src/smt/command.cpp:702]: Why is it called here rather than when the command is declared?

Comment (ajreynol) [reply src/smt/command.cpp:702]: Creating a command should not do this.  It is only if/when the command is invoked that it should update the state of symbol manager.

Comment (HanielB) [reply src/smt/command.cpp:702]: I see, cool.

**Review by HanielB — APPROVED** [review]

**Pull request merged**